### ISSUE TITLE
Different result of model.predict(X) and np.argmax(model.predict_proba(X_val), axis=1)

### ISSUE THREAD

**Issue by zw-SIMM** [PriorLabs/tabpfn-extensions#101]
### Describe the bug

When I using many_class， the result of model.predict(X) and np.argmax(model.predict_proba(X_val), axis=1) is different

```python
X_train, X_val = X[train_index], X[val_index]
y_train, y_val = y[train_index], y[val_index]

base_clf = TabPFNClassifier(random_state=42, ignore_pretraining_limits=True)
model = ManyClassClassifier(base_clf, alphabet_size=10)
model.fit(X_train, y_train)

y_pred = model.predict(X_val)
```

Output is 
```shell
array([ 46,  46,  48,  48,  48,  48,  48,  48,  48,  48, 143, 139, 185,
       185, 168, 168, 168, 168, 161, 168, 161,  36,  36,  36,  36,  36,
        36,  60,  36,  94, 110, 110, 110,   2,   2,   2,   2,   2,  12,
         2,   2, 168, 168, 168, 168,  48,  48,  48,  48,  48, 168, 168,
       168, 168, 168, 173, 168,   6,   6,   6,   6,   6, 122,   6, 122,
       122, 122,  48,  48,  48,  48,  48,  48,  48,  48,  48, 160, 160,
        48,  48,  48,  48,  48,  48,  48,  48,  48,  48,  48, 160,  42,
        42,  42,  42,  42,  42,  61,  61,  23,  23,  23, 102, 102, 184,
       128, 118, 118,  75,  75, 131, 131, 131, 131,  18,  18,  18,  18,
        18,  18,  18,  18,  18,  88, 168, 168, 168,  48, 162, 162, 156,
       156, 140, 140, 140, 140, 140, 140, 140, 133, 133, 133, 133, 133,
       133, 185, 185, 185, 185,  46,  46,  46,  46,  46,  46,  46,  46,
       156, 156,  62,  62,  65,  65,  65,  65,  65,  65,  62,  62,   6,
       142, 101, 101, 144, 144, 144, 144, 144, 144, 142, 117,  61,  61,
        61,  61,  61,  61,  61,  61,  61,  61,  16,  18,  16,  16,  18,
        18,  18,  46,  46,  46,  46,  46,  46,  46,  46,  43,  43,  43,
        43,  43,  43,  43,  42,  42,  42,  42,  42,  42,  42,  42,  42,
        42,  42,  42,  43,  42,  18,  16,  16, 169, 169, 169, 169, 149,
       149, 149, 149, 149, 149, 149, 184, 184, 184,  88,  88,  89, 168,
       168, 168, 168, 160, 168, 160, 165, 168, 168, 168,  10,  10,  10,
        10,  10,  10,  10,  10,  10,  10,  10,  28,  35, 176, 176, 176,
       170, 170,  30,  30, 170, 147, 147, 147, 116, 116, 130, 130, 130,
       130, 130, 130, 130,  48,  48,  48,  48,  48,  48,  48,  48,  48,
        48,  72,  72, 177, 177,  72,  72,  72,  72, 177,  72, 175, 175,
       174, 174, 174, 174, 174, 174, 174, 174, 182, 182, 174, 175, 175,
       174, 170, 174, 174, 145, 145, 128, 128, 128,  40, 145, 145, 127,
       170, 127, 127, 174,  38,  36,  25, 174, 174,  36,  36,  36,  36,
        57,  36, 174,  40,  63,   8,   8,   8,   8,   8,   8,   8,   8,
        38,   8,   8,   8,  38,  36,  36,  36,  36,  36, 135,  57,  57,
        63,  63,  63,  63,  63, 128, 128,  40, 104, 104, 168,   9])
```

```python
np.argmax(model.predict_proba(X_val), axis=1)
```

output is
```shell
array([ 40,  40,  42,  42,  42,  42,  42,  42,  42,  42, 129, 125, 169,
       169, 152, 152, 152, 146, 145, 152, 145,  31,  31,  31,  31,  31,
        31,  31,  31,  83,  99,  99,  99,   2,   2,   2,   2,   2,   0,
         0,   2, 152, 152, 152, 152,  42,  42,  42,  42,  42, 152, 152,
       152, 152, 152, 128, 152,   5,   5,   5,  50,   5, 110,  69, 110,
       110, 110,  42,  42,  42,  42,  42,  42,  42,  42,  42, 144, 144,
        42,  42,  42,  42,  42,  42,  42,  42,  42,  42,  42, 144,  36,
        36,  36,  36,  36,  36,  52,  52,  20,  20,  20,  91,  91, 168,
       168, 106, 106,  66,  66, 118, 118, 118, 118,  15,  15,  15,  15,
        15,  15,  15,  15,  15,  77, 152, 152, 152,  77, 152,  69, 140,
       140, 126, 126, 126, 126, 126, 126, 126, 120, 120, 120, 120, 120,
       120, 169, 169, 169, 169,  40,  40,  40,  40,  40,  40,  40,  40,
       140, 140,  53,  53,  56,  56,  56,  56,  56,  36,  53,  53,  90,
        69,  90,  90, 130, 130, 130, 130, 130, 130, 128,  69,  52,  52,
        52,  52,  52,  52,  52,  52,  52,  52,  15,  15,  13,  13,  15,
        15,  15,  40,  40,  40,  40,  40,  40,  40,  40,  37,  37,  37,
        37,  37,  37,  37,  36,  36,  36,  36,  36,  36,  36,  36,  36,
        69,  69,  36,  37,  23,  15,  13,  13, 153, 153, 153, 153, 135,
       135, 135, 135, 135, 135, 152, 168, 168, 168,  77,  77,  59, 145,
       144, 152, 144, 144, 144, 144, 128, 152, 145, 152,   8,   8,   8,
         8,   8,   8,   8,   8,   8,   8,   8,   8, 165, 160, 160, 160,
        69,  69,  69,  69,  69, 133, 133, 133, 104, 104, 117, 117, 117,
       117, 117, 117, 117,  42,  42,  42,  42,  42,  42,  42,  42,  42,
        42,  63,  63, 161, 161,  63,  63,  63,  63, 161,  63, 159, 159,
       158, 158, 158, 158, 158, 158, 158, 158, 166, 166, 158, 159, 159,
       158,  21, 158, 158, 131, 131, 116, 116, 116,  34, 131, 131, 115,
       115, 115, 115, 158,  32,  31,  22, 158, 158,  31,  31,  31,  31,
        48,  31,  31,  34,  54,   6,   6,   6,   6,   4,   4,   6,   6,
        32,   6,   6,   6,  32,  31,  31,  31,  31,  31, 123,  48,  48,
        54,  54,  54,  54,  54, 116, 116,  34,  93,  93, 121,   7])
```

### Versions

```shell
tabpfn                    2.0.9
tabpfn-extensions         0.1.0
```
